module/zfs/zfs_ctldir.c

```diff
@@ -492,6 +492,9 @@ zfsctl_inode_alloc(zfsvfs_t *zfsvfs, uint64_t id,
 	ip->i_ctime = now;
 	ip->i_fop = fops;
 	ip->i_op = ops;
+#ifndef HAVE_GENERIC_SETXATTR
+	ip->i_opflags = ~IOP_XATTR;
+#endif
```

Comment:
Nice find, yes we should definitely do this but I'd suggest adding this same logic a little differently.  The following has a few advantages.

* Checking for `IOP_XATTR` instead of `HAVE_GENERIC_SETXATTR` ensures we're depending on exactly this interface change.  That way in the unlikely case this change gets cherry picked in to an older kernel this will still work correctly.

* Using `&= ~IOP_XATTR` just clears the one bit we care about.

```c
#if defined(IOP_XATTR)
        ip->i_opflags &= ~IOP_XATTR;
#endif
```